### PR TITLE
CI: Jenkins stage with test cases for QAT and FPGA

### PR DESCRIPTION
**Pull request by hectoragb**
This PR adds FPGA tests to the QAT tests, see https://github.com/intel/intel-device-plugins-for-kubernetes/pull/231. It also replaces https://github.com/intel/intel-device-plugins-for-kubernetes/pull/235 where the https://github.com/intel/intel-device-plugins-for-kubernetes/pull/231 is squashed

Signed-off-by: Hector Augusto Garcia Baleon <hector.augusto.garcia.baleon@intel.com>

### CODE REVIEW

**Comment by hectoragb** [comment]
@mythi This PR replaces https://github.com/intel/intel-device-plugins-for-kubernetes/pull/235 and https://github.com/intel/intel-device-plugins-for-kubernetes/pull/231. Changes are in line with the BMaaS which has been setup and tested already.

File: demo/test-fpga-region.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ spec:
           [IPC_LOCK]
     resources:
       limits:
-        fpga.intel.com/arria10.dcp1.2-nlb3: 1
+        fpga.intel.com/arria10.dcp1.1-nlb3: 1
```
Comment (bart0sh) [inline demo/test-fpga-region.yaml]: Can you explain why we downgrade DCP version here? It makes sense to use the latest version available I believe.

File: scripts/jenkins/fpga/nlb-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#
+# Deploys current jenkins-build test image for nlb-demo
+# in the cluster.
+
+set -o pipefail
+set -o xtrace
+set -o errexit
+
+REPO_ROOT=$(realpath $(dirname $0)/../../..)
+kubectl create -f ${REPO_ROOT}/demo/test-fpga-region.yaml
```
Comment (bart0sh) [inline scripts/jenkins/fpga/nlb-deploy.sh]: Are there any plans to test FPGA in AF mode?

**Comment by mythi** [comment]
@hectoragb I'd prefer to keep these separate PRs. You can keep #235 as the base but add your FPGA commits separately.

File: scripts/jenkins/fpga/nlb-deploy.sh (hunk 2 above)
Comment (hectoragb) [reply scripts/jenkins/fpga/nlb-deploy.sh]: Yes. Eventually, aocx testing, as well. More testing is needed on the backend K8s nodes

File: demo/test-fpga-region.yaml (hunk 1 above)
Comment (hectoragb) [reply demo/test-fpga-region.yaml]: I focused on having a reliable K8s infrastructure. I have not upgraded all FPGA cards to DCP 1.2 so I will change accordingly the DCP version once the upgrading step is ready and tested

**Pull request closed without merging**